src/services/weatherService.js

```diff
@@ -0,0 +1,53 @@
+import { getDayOfTheWeekName, getDateTime } from '../utils/dateUtils';
+import { API_KEY } from '../secrets';
+import { kelvinToCelsius } from '../utils/weatherUtils';
+
+const convertToCelsiusAndFloor = (t) => Math.floor(kelvinToCelsius(t));
+
+export const getTemperaturesByCityNames = (cityNames) =>
+  Promise.all(
+    cityNames.map((cityName) =>
+      fetch(`http://api.openweathermap.org/data/2.5/weather?q=${cityName}&APPID=${API_KEY}`)
+    )
+  )
+    .then((responses) => Promise.all(responses.map((response) => response.json())))
+    .then((weathers) =>
+      weathers.map((weather) => ({
+        name: weather.name,
+        temperature: convertToCelsiusAndFloor(weather.main.temp),
+      }))
+    );
+
+export const getWeatherByGeolocation = ({ coords: { latitude, longitude } }) =>
+  fetch(
+    `http://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&APPID=${API_KEY}`
+  )
+    .then((resp) => resp.json())
+    .then(async (weather) => {
```

Comment:
Just curious: why did you decide to mix async/await with promises ?

Reply:
After I receive `weather ` info from api.openweathermap.org I want to get full `country.name`. How would you do that?